src/widget/textbox.rs

```diff
@@ -169,25 +171,16 @@ impl Widget for TextBox {
     }
     true
   }
+
   fn key(&mut self, event: &KeyEvent, ctx: &mut HandlerCtx) -> bool {
     //match on key event
-    match event.key {
-      KeyVariant::Char(ch) => {
-        if ch == '\u{7f}' {
-          self.text.pop();
-        } else {
-          self.text.push(ch);
-        }
-      }
-      KeyVariant::Vkey(vk) => match vk {
-        VK_BACK => {
+    match event {
+      KeyEvent::Character(data) => self.text.push_str(data.text().unwrap_or("")),
+      KeyEvent::NonCharacter(data) if data.key_code == KeyCode::Backspace => {
           self.text.pop();
-        }
-        _ => {}
       },
+      _other => return false,
```

Comment:
I personally feel just plain `_` is idiomatic and shorter, but that's getting deep into style nits; feel free to keep it if you like it.